apps/web/src/components/v2/manager-chat.tsx

```diff
@@ -135,21 +159,36 @@ export function ManagerChat({
             onKeyDown={(e) => {
               if (e.key === "Enter" && (e.metaKey || e.ctrlKey)) {
                 e.preventDefault();
-                handleSendMessage();
+                if (stream.isLoading) {
+                  handleCancel();
+                } else {
+                  handleSendMessage();
+                }
               }
             }}
           />
           <Button
-            onClick={handleSendMessage}
-            disabled={!chatInput.trim()}
-            size="icon"
-            variant="brand"
+            onClick={stream.isLoading ? handleCancel : handleSendMessage}
+            disabled={stream.isLoading ? false : !chatInput.trim()}
+            size={stream.isLoading ? "sm" : "icon"}
+            variant={stream.isLoading ? "destructive" : "brand"}
+            className={cn(
+              "self-end transition-all duration-200 disabled:opacity-50",
+              stream.isLoading ? "h-12 px-4 py-2" : "h-10 w-10 p-0",
+            )}
```

Comment:
we didnt have any `className` prop passed before, so we should only pass a custom `className` if we're showing the new loader icon & cancel text
```suggestion
            className={cn(
              stream.isLoading ? "h-12 px-4 py-2" : "",
            )}
```

(not sure if this `className` will work as intended tho, so double check locally)